src/LinAlg/hiopMatrixDense.hpp

```diff
@@ -214,12 +214,12 @@ class hiopMatrixDense : public hiopMatrix
   virtual long long get_local_size_m() const {assert(false && "not implemented in base class"); return -1;}
   virtual MPI_Comm get_mpi_comm() const { return comm_; }
 
-  //TODO: this is not kosher!
-  virtual double** local_data() const {assert(false && "not implemented in base class"); return nullptr;}
-  virtual double*  local_buffer() const {assert(false && "not implemented in base class"); return nullptr;}
+  virtual double* local_data_const() const {assert(false && "not implemented in base class"); return nullptr;}
+  virtual double* local_data() {assert(false && "not implemented in base class"); return nullptr;}
+protected:
   //do not use this unless you sure you know what you're doing
   virtual double** get_M(){assert(false && "not implemented in base class"); return nullptr;}
```

Comment:
```suggestion
```
Probably a good idea to remove this altogether.

Reply:
removed from dense base and raja dense classes. kept in rowmajor dense.